logic/resolvable/RetrievableTest.java

```diff
@@ -158,10 +158,10 @@ public void test_retrievables_split_by_labelled_type_are_separated() {
     }
 
     @Test
-    public void test_has_value_constraints_are_in_retrievable_and_concludable() {
-        Set<Concludable> concludables = Concludable.create(parse("{ $a has $b; $b > 5; $b < 10; }"));
+    public void test_has_value_constraints_are_in_retrievable_iff_they_are_not_in_concludable() {
+        Set<Concludable> concludables = Concludable.create(parse("{ $a has $b; $b > 5; }"));
         Set<Retrievable> retrievables = Retrievable.extractFrom(parse("{ $a has $b; $b > 5; $b < 10; }"), concludables);
-        assertEquals(set(parse("{ $b > 5; $b < 10; }")),
+        assertEquals(set(parse("{ $b < 10; }")),
```

Comment:
Since we can now handle constraint compatibility, AND evaluate any constant predicates in the Unifier, we can move all the constant value constraints into the single concludable exclusively.

This should be more efficient because we create potentially fewer unifiers and also potentially eliminate some Retrievables totally.

Verify that `$a has $b; $c has $b; $b > 10;` ends up with two concludables and 0 retrievables: `$a has $b; $b > 10;` and `$c has $b; $b > 10;`